airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/staging/StagingConsumerFactory.java

```diff
@@ -157,16 +155,17 @@ private RecordWriter recordWriterFunction(final JdbcDatabase database,
       final WriteConfig writeConfig = pairToWriteConfig.get(pair);
       final String schemaName = writeConfig.getOutputSchemaName();
       final String tableName = writeConfig.getOutputTableName();
-      final String path = namingResolver.getStagingPath(schemaName, tableName, CURRENT_SYNC_PATH);
+      final String stage = stagingOperations.getStageName(schemaName, tableName);
+      final String path = stagingOperations.getStagingPath(RANDOM_CONNECTION_ID, schemaName, tableName, CURRENT_SYNC_PATH);
 
-      snowflakeSqlOperations.insertRecords(database, records, schemaName, path);
+      stagingOperations.createStageIfNotExists(database, stage);
```

Comment:
isn't this already happening in onStartFunction? (it looks like this was to fix a test, but I'm not sure I understand why this was the fix)

Reply:
You're right!

I was just on the fence on whether to create the stage in the `onStart` or `onRecordWrite` phase...